standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -1636,6 +1637,10 @@ public static String getPartitionName(Path tablePath, Path partitionPath, Set<St
 
   public static String getNormalisedPartitionValue(String partitionValue, String type) {
 
+    if (!NumberUtils.isParsable(partitionValue)) {
+      return partitionValue;
+    }
```

Comment:
I slightly suspect this can skip the case of `decimal`.
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#isParsable(java.lang.String)

Reply:
@okumin , thanks for the input. I think there is already a UT for this and passing post the patch. Please check https://github.com/apache/hive/blob/0e75930d619e96a32ae35b59df9eeeb6466ea2c6/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreServerUtils.java#L965

Let me know if I am missing something, but **NumberUtils.isParsable(partitionValue)** will return true for a decimal and later in same function it will be handled. https://github.com/apache/hive/blob/0e75930d619e96a32ae35b59df9eeeb6466ea2c6/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L1653

Reply:
I originally thought it could not work when `Double.parseDouble` can not decode it but `new BigDecimal` can decode it. An example in my mind was a huge number. Looking at the implementation, now I feel the current implementation would work
https://github.com/apache/commons-lang/blob/daa63aa595ea39a69e1b44ac23bc50fdcf2afd62/src/main/java/org/apache/commons/lang3/math/NumberUtils.java#L1794-L1810